packages/dev-server/src/utils/getIPV4.ts

```diff
@@ -0,0 +1,32 @@
+import os from 'os';
+
+const localIpv4 = '127.0.0.1';
+
+const getIpv4Interfaces = (): os.NetworkInterfaceInfo[] => {
+  try {
+    const interfaces = os.networkInterfaces();
+    const ipv4Interfaces: os.NetworkInterfaceInfo[] = [];
+
+    Object.keys(interfaces).forEach((key) => {
+      interfaces[key]!.forEach((detail) => {
+        // 'IPv4' is in Node <= 17, from 18 it's a number 4 or 6
+        const familyV4Value = typeof detail.family === 'string' ? 'IPv4' : 4;
+
+        if (detail.family === familyV4Value && detail.address !== localIpv4) {
+          ipv4Interfaces.push(detail);
+        }
+      });
+    });
```

Comment:
Instead of taking the keys of the object, just take directly its values with Object.values